### PR TITLE
Chain Fusion Docs: Ethereum -> ICP for ETH Devs

### PR DESCRIPTION
**Pull request by jennifertrin**
Thank you for your contribution to the IC Developer Portal. This repo contains the content for https://internetcomputer.org and the ICP Developer Documentation, https://internetcomputer.org/docs/. 

If you are submitting a Pull Request for adding or changing content on the ICP Developer Documentation, please make sure that your contribution meets the following requirements:

- [X] Follows the [developer docs style guide](style-guide.md).
- [X] Follows the [best practices and guidelines](README.md#best-practices).
- [X] New documentation pages include [document tags](README.md#document-tags).
- [X] New documentation pages include [SEO keywords](README#seo-keywords).
- [X] New documents are in the `.mdx` file format to support the previous two components. 
- [X] New documents are registered in [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js), otherwise, it will not appear in the
  side navigation bar.
- [ ] Make sure that the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file is
  filled with new documents that you added. This way we can ensure that future Pull Requests are reviewed by the right people.

### CODE REVIEW

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+keywords: [intermediate, ethereum, eth, icp]
+---
+```
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:4]:
```suggestion

```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Understanding ICP for Ethereum developers
+
+<MarkdownChipRow labels={["Intermediate", "Ethereum"]} />
+````
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:10]:
```suggestion

```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## Overview
+This guide is designed to help Ethereum developers understand the core concepts of ICP, focusing on the terminology and functionalities that differ between the two protocols. While the underlying concepts might be similar, ICP utilizes different terms to describe them.
+
+For a feature comparison between the two protocols, refer to the [Quick Comparison with ICP and Ethereum](../../../../developer-docs/getting-started/overview-of-icp#quick-comparison-with-ethereum).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:15]:
```suggestion
For a feature comparison between the two protocols, refer to the [quick comparison between ICP and Ethereum](/docs/current/developer-docs/getting-started/overview-of-icp#quick-comparison-with-ethereum).
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+## Terminology mapping
+This table summarizes the key terminology differences between Ethereum and ICP:
+
+<table>
+<thead>
+<tr>
+  <th> Ethereum Term </th>
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:23]:
```suggestion
  <th> Ethereum term </th>
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  <th> ICP Term </th>
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:24]:
```suggestion
  <th> ICP term </th>
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+</tr>
+
+<tr>
+  <td> Account </td>
+  <td> Principal </td>
+</tr>
+
+<tr>
+  <td> DAO </td>
+  <td> SNS (ICP has a specific DAO framework named an [SNS](../../../../developer-docs/daos/sns/). However, DAOs can launch without using this framework) </td>
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:56]:
```suggestion
  <td> SNS, a DAO using ICP's [unique framework](/docs/current/developer-docs/daos/sns/). However, ICP DAOs can launch without using this framework. </td>
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+</tr>
+
+## Smart contract development comparison
+
+Both Ethereum and ICP allow developers to deploy programs that run on the blockchain. These programs are referred to as smart contracts in Ethereum and [canisters](../../../../developer-docs/smart-contracts/overview/) on ICP. Occasionally, the phrase “canister smart contracts” is also used on ICP.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:61]:
```suggestion
Both Ethereum and ICP allow developers to deploy programs that run on the blockchain. These programs are referred to as smart contracts in Ethereum and smart contract canisters, or simply just [canisters],(/docs/current/developer-docs/smart-contracts/overview/) on ICP. 
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+This comparison provides an overview of the key differences in smart contract development on Ethereum and ICP.
+
+## Key differences:
+
+### Creation:
+**Ethereum**: Smart contracts are written in Solidity and deployed immutably onto the blockchain.
+**ICP**: Canisters can be written in any language that complies to WASM including [Rust](../../../../developer-docs/backend/rust/), [Typescript](../../../../developer-docs/backend/Typescript), [Motoko](../../../../developer-docs/backend/motoko/), and [Python](../../../../developer-docs/backend/python/).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:69]:
```suggestion
## Key differences

### Creation
**Ethereum**: Smart contracts are written in Solidity and deployed immutably onto the blockchain.
**ICP**: Canisters can be written in any language that complies to Wasm including [Rust](/docs/current/developer-docs/backend/rust/), [Typescript](/docs/current/developer-docs/backend/Typescript), [Motoko](/docs/current/developer-docs/backend/motoko/), and [Python](/docs/current/developer-docs/backend/python/).
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+### Paying for gas
+
+**Ethereum**: Users pay for gas fees directly using the crypto wallet they connect with the dapp.
+**ICP**: Using the [reverse gas model](../../../../developer-docs/gas-cost/#the-reverse-gas-model), the dapp developers, not users, pay for the gas fees. The reverse gas model aims to provide a smoother user experience by eliminating upfront fees.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:74]:
```suggestion
**ICP**: Using the [reverse gas model](/docs/current/developer-docs/gas-cost/#the-reverse-gas-model), the dapp developers, not the users, pay for the gas fees. The reverse gas model aims to provide a smoother user experience by eliminating upfront fees.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+### Updates:
+**Ethereum**: Smart contracts are immutable. To update functionalities, developers need to create a new smart contract and migrate data.
+**ICP**: Developers can create upgradeable or non-upgradable (also known as "blackhole") canisters. Upgradable canisters allow updates by reinstalling the code or deploying a new version. Non-upgradable canisters are similar to Ethereum smart contracts.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:78]:
```suggestion
### Updates
**Ethereum**: Smart contracts are immutable. To update functionalities, developers need to create a new smart contract and migrate data.
**ICP**: Developers can create upgradeable or non-upgradable (also known as "blackhole") canisters. Upgradable canisters allow updates by reinstalling the code or deploying a new version. Non-upgradable canisters are similar to Ethereum smart contracts.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+### Testing
+**Ethereum**: Developers can utilize testnets for deploying and testing smart contracts before deploying to mainnet.
+
+**ICP**: Currently, testing is done in the local environment. A [playground environment](../../../developer-tools/ide/playground.mdx) is available for canisters written into Motoko. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their canisters.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:83]:
```suggestion
### Testing

**Ethereum**: Developers can utilize testnets such as Sepolia for deploying and testing smart contracts before deploying to the mainnet.
**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-tools/ide/playground.mdx) is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their canisters.
```

Comment (jessiemongeon1) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:83]: The Motoko Playground supports canisters in any language, not just Motoko (despite its name)

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx (hunk 8 above)
Comment (jessiemongeon1) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:69]: Should we include an example Solidity 'Hello, world!' contract and an example Rust/Motoko 'Hello, world!' canister to show a side-by-side of the syntax?

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+Both Ethereum and ICP allow developers to deploy programs that run on the blockchain. These programs are referred to as smart contracts in Ethereum and [canisters](../../../../developer-docs/smart-contracts/overview/) on ICP. Occasionally, the phrase “canister smart contracts” is also used on ICP.
+
+This comparison provides an overview of the key differences in smart contract development on Ethereum and ICP.
+
+## Key differences:
+
+### Creation:
+**Ethereum**: Smart contracts are written in Solidity and deployed immutably onto the blockchain.
+**ICP**: Canisters can be written in any language that complies to WASM including [Rust](../../../../developer-docs/backend/rust/), [Typescript](../../../../developer-docs/backend/Typescript), [Motoko](../../../../developer-docs/backend/motoko/), and [Python](../../../../developer-docs/backend/python/).
+
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:70]: We should add a 'Deployment' section that details how Ethereum uses tools like Hardhat/Truffle for deployment while ICP uses dfx.

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+### Paying for gas
+
+**Ethereum**: Users pay for gas fees directly using the crypto wallet they connect with the dapp.
+**ICP**: Using the [reverse gas model](../../../../developer-docs/gas-cost/#the-reverse-gas-model), the dapp developers, not users, pay for the gas fees. The reverse gas model aims to provide a smoother user experience by eliminating upfront fees.
+
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:75]: Should we add a section on token standards and mention ERC20 on Ethereum, then ICRC-1/2 on ICP?

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx (hunk 8 above)
Comment (letmejustputthishere) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:67]: i think that's a great idea. small nitpick, typescript and python aren't compiled to Wasm. they have an interpreter written in rust that can be compiled to wasm. the interpreter then runs within the canister and interprets the js/python canister code.

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx (hunk 12 above)
Comment (letmejustputthishere) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:68]: we could also highlight that `scaffold-eth` is similar to our `dfx new` command

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx (hunk 13 above)
Comment (letmejustputthishere) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:73]: great idea, we can also add ICRC7 vs ERC721

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+**ICP**: Canisters can be written in any language that complies to WASM including [Rust](../../../../developer-docs/backend/rust/), [Typescript](../../../../developer-docs/backend/Typescript), [Motoko](../../../../developer-docs/backend/motoko/), and [Python](../../../../developer-docs/backend/python/).
+
+### Paying for gas
+
+**Ethereum**: Users pay for gas fees directly using the crypto wallet they connect with the dapp.
+**ICP**: Using the [reverse gas model](/docs/current/developer-docs/gas-cost/#the-reverse-gas-model), the dapp developers, not the users, pay for the gas fees. The reverse gas model aims to provide a smoother user experience by eliminating upfront fees.
+
+### Updates
+**Ethereum**: Smart contracts are immutable. To update functionalities, developers need to create a new smart contract and migrate data.
+**ICP**: Developers can create upgradeable or non-upgradable (also known as "blackhole") canisters. Upgradable canisters allow updates by reinstalling the code or deploying a new version. Non-upgradable canisters are similar to Ethereum smart contracts.
```
Comment (letmejustputthishere) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:76]:
```suggestion
**ICP**: Developers can create upgradeable or non-upgradable (also known as "blackholed") canisters. Upgradable canisters allow updates by reinstalling the code or deploying a new version. Non-upgradable canisters are similar to Ethereum smart contracts.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+### Testing
+
+**Ethereum**: Developers can utilize testnets such as Sepolia for deploying and testing smart contracts before deploying to the mainnet.
+**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-tools/ide/playground.mdx) is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their canisters.
```
Comment (letmejustputthishere) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:81]:
```suggestion
**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-tools/ide/playground.mdx) on mainnet is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their canisters.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx (hunk 13 above)
Comment (jennifertrin) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:73]: Thank you for the idea! I will add by EOD tomorrow.

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx (hunk 12 above)
Comment (jennifertrin) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:68]: Thank you for the idea! I will add by EOD tomorrow.

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+
+Both Ethereum and ICP allow developers to deploy programs that run on the blockchain. These programs are referred to as smart contracts in Ethereum and [canisters](/docs/current/developer-docs/smart-contracts/overview/introduction) on ICP. Occasionally, the phrase “canister smart contracts” is also used on ICP.
+
+This comparison provides an overview of the key differences in smart contract development on Ethereum and ICP.
+
+## Key differences:
+
+### Creation:
+**Ethereum**: Smart contracts are written in Solidity and deployed immutably onto the blockchain.
+**ICP**: Canisters can be written in any language that complies to WASM including [Rust](/docs/current/developer-docs/backend/rust/), [Typescript](/docs/current/developer-docs/backend/Typescript), [Motoko](/docs/current/motoko/main/getting-started/motoko-introduction), and [Python](/docs/current/developer-docs/backend/python/).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:92]:
```suggestion
**ICP**: Canisters can be written in any language that complies to Wasm including [Rust](/docs/current/developer-docs/backend/rust/), [Typescript](/docs/current/developer-docs/backend/Typescript), [Motoko](/docs/current/motoko/main/getting-started/motoko-introduction), and [Python](/docs/current/developer-docs/backend/python/).
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+
+### Deployment
+**Ethereum**: Developers typically use development frameworks like Hardhat or Truffle to compile and deploy their smart contracts. 
+**ICP**: Developers use dfx which is a similar development environment. [DFX](/docs/current/developer-docs/getting-started/install/#installing-dfx-via-dfxvm) is a command-line tool that interacts with the IC SDK and allows developers to compile and deploy their canisters.
+
+### Testing
+
+**Ethereum**: Developers can utilize testnets such as Sepolia for deploying and testing smart contracts before deploying to the mainnet.
+**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-docs/developer-tools/ide/playground/) on mainnet is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their canisters.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:101]:
```suggestion
**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-docs/developer-tools/ide/playground/) on mainnet is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility thank on Ethereum to iterate and refine their canisters.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+**Ethereum**: Users pay for gas fees directly using the crypto wallet they connect with the dapp.
+**ICP**: Using the [reverse gas model](/docs/current/developer-docs/gas-cost/#the-reverse-gas-model), the dapp developers, not the users, pay for the gas fees. The reverse gas model aims to provide a smoother user experience by eliminating upfront fees.
+
+### Updates
+**Ethereum**: Smart contracts are immutable. To update functionalities, developers need to create a new smart contract and migrate data.
+**ICP**: Developers can create upgradeable or non-upgradable (also known as "blackholed") canisters. Upgradable canisters allow updates by reinstalling the code or deploying a new version. Non-upgradable canisters are similar to Ethereum smart contracts.
+
+### Consensus
+
+***Ethereum**: Ethereum uses [Proof of Stake](https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/) as its consensus mechanism.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:114]:
```suggestion
***Ethereum**: Ethereum uses [proof of stake](https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/) as its consensus mechanism.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+***ICP**: The IC consensus protocol provides cryptographically guaranteed finality. The approach on ICP avoids the drawbacks of probabilistic finality used in Bitcoin-like protocols, which include the weak notion and increased time to finality. The consensus mechanism of ICP ensures faster and more secure transaction finality.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:115]:
```suggestion
***ICP**: The IC consensus protocol provides cryptographically guaranteed finality. The approach on ICP solves the drawbacks of probabilistic finality used in Bitcoin-like protocols, including slow finality times. ICP ensures faster and more secure transaction finality.
```

**Review by jessiemongeon1 — APPROVED** [review]

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 20]
```diff
@@ -0,0 +1,124 @@
...
+
+## Key differences:
+
+### Creation:
+**Ethereum**: Smart contracts are written in Solidity and deployed immutably onto the blockchain.
+**ICP**: Canisters can be written in any language that complies to WASM including [Rust](/docs/current/developer-docs/backend/rust/), [Typescript](/docs/current/developer-docs/backend/Typescript), [Motoko](/docs/current/motoko/main/getting-started/motoko-introduction), and [Python](/docs/current/developer-docs/backend/python/).
+
+### Deployment
+**Ethereum**: Developers typically use development frameworks like Hardhat or Truffle to compile and deploy their smart contracts. 
+**ICP**: Developers use dfx which is a similar development environment. [DFX](/docs/current/developer-docs/getting-started/install/#installing-dfx-via-dfxvm) is a command-line tool that interacts with the IC SDK and allows developers to compile and deploy their canisters.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:96]:
```suggestion
**ICP**: Developers use `dfx` which is a similar development environment. [`dfx`](/docs/current/developer-docs/getting-started/install/#installing-dfx-via-dfxvm) is a command-line tool that interacts with the IC SDK and allows developers to compile and deploy their canisters.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 21]
```diff
@@ -0,0 +1,124 @@
...
+</tr>
+
+<tr>
+  <td> ERC-721 </td>
+  <td> [ICRC-7](https://github.com/dfinity/ICRC/blob/icrc7-wg-draft/ICRCs/ICRC-7/ICRC-7.md) </td>
+</tr>
+</tbody>
+</table>
+
+## Smart contract development comparison
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:82]:
```suggestion
## Smart contracts
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 22]
```diff
@@ -0,0 +1,124 @@
...
+
+Both Ethereum and ICP allow developers to deploy programs that run on the blockchain. These programs are referred to as smart contracts in Ethereum and [canisters](/docs/current/developer-docs/smart-contracts/overview/introduction) on ICP. Occasionally, the phrase “canister smart contracts” is also used on ICP.
+
+This comparison provides an overview of the key differences in smart contract development on Ethereum and ICP.
+
+## Key differences:
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:88]:
```suggestion
### Key differences
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+
+### Creation:
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:90]:
```suggestion
#### Creation
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 24]
```diff
@@ -0,0 +1,124 @@
...
+**Ethereum**: Smart contracts are written in Solidity and deployed immutably onto the blockchain.
+**ICP**: Canisters can be written in any language that complies to WASM including [Rust](/docs/current/developer-docs/backend/rust/), [Typescript](/docs/current/developer-docs/backend/Typescript), [Motoko](/docs/current/motoko/main/getting-started/motoko-introduction), and [Python](/docs/current/developer-docs/backend/python/).
+
+### Deployment
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:94]:
```suggestion
#### Deployment
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 25]
```diff
@@ -0,0 +1,124 @@
...
+**Ethereum**: Developers typically use development frameworks like Hardhat or Truffle to compile and deploy their smart contracts. 
+**ICP**: Developers use dfx which is a similar development environment. [DFX](/docs/current/developer-docs/getting-started/install/#installing-dfx-via-dfxvm) is a command-line tool that interacts with the IC SDK and allows developers to compile and deploy their canisters.
+
+### Testing
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:98]:
```suggestion
#### Testing
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 26]
```diff
@@ -0,0 +1,124 @@
...
+
+**Ethereum**: Developers can utilize testnets such as Sepolia for deploying and testing smart contracts before deploying to the mainnet.
+**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-docs/developer-tools/ide/playground/) on mainnet is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their canisters.
+
+### Paying for gas
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:103]:
```suggestion
#### Paying for gas
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 27]
```diff
@@ -0,0 +1,124 @@
...
+
+**Ethereum**: Users pay for gas fees directly using the crypto wallet they connect with the dapp.
+**ICP**: Using the [reverse gas model](/docs/current/developer-docs/gas-cost/#the-reverse-gas-model), the dapp developers, not the users, pay for the gas fees. The reverse gas model aims to provide a smoother user experience by eliminating upfront fees.
+
+### Updates
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:108]:
```suggestion
#### Updates
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 28]
```diff
@@ -0,0 +1,124 @@
...
+**Ethereum**: Smart contracts are immutable. To update functionalities, developers need to create a new smart contract and migrate data.
+**ICP**: Developers can create upgradeable or non-upgradable (also known as "blackholed") canisters. Upgradable canisters allow updates by reinstalling the code or deploying a new version. Non-upgradable canisters are similar to Ethereum smart contracts.
+
+### Consensus
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:112]:
```suggestion
## Consensus
```

**Comment by jessiemongeon1** [comment]
Some community feedback from the forum that might be good to include in this page:

"I don't see the concept of contract accounts and message calls that we have in Solidity"
"[What are the ICP equivalents of] ERC721 contract, ICO token, contract for Uniswap V3 protocol?"

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx (hunk 17 above)
Comment (jennifertrin) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:101]: @jessiemongeon1 Can you please take a look at this suggestion?

Comment (jessiemongeon1) [reply docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:101]: Updated suggestion

**Comment by jennifertrin** [comment]
> Some community feedback from the forum that might be good to include in this page:
> 
> "I don't see the concept of contract accounts and message calls that we have in Solidity" "[What are the ICP equivalents of] ERC721 contract, ICO token, contract for Uniswap V3 protocol?"

Thank you for the feedback. I will add by EOW

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 29]
```diff
@@ -0,0 +1,124 @@
...
+<tr>
+  <td> ERC-721 </td>
+  <td> [ICRC-7](https://github.com/dfinity/ICRC/blob/icrc7-wg-draft/ICRCs/ICRC-7/ICRC-7.md) </td>
+</tr>
+</tbody>
+</table>
+
+## Smart contracts
+
+Both Ethereum and ICP allow developers to deploy programs that run on the blockchain. These programs are referred to as smart contracts in Ethereum and canister smart contracts, or simply just [canisters](/docs/current/developer-docs/smart-contracts/overview/) on ICP.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:84]:
```suggestion
Both Ethereum and ICP allow developers to deploy programs that run on-chain. These programs are referred to as smart contracts on Ethereum and canister smart contracts, or simply just [canisters](/docs/current/developer-docs/smart-contracts/overview/), on ICP.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 30]
```diff
@@ -0,0 +1,124 @@
...
+
+#### Deployment
+**Ethereum**: Developers typically use development frameworks like Hardhat or Truffle to compile and deploy their smart contracts. 
+**ICP**: Developers use `dfx` which is a similar development environment. [`dfx`](/docs/current/developer-docs/getting-started/install/#installing-dfx-via-dfxvm) is a command-line tool that interacts with the IC SDK and allows developers to compile and deploy their canisters.
+
+#### Testing
+
+**Ethereum**: Developers can utilize testnets such as Sepolia for deploying and testing smart contracts before deploying to the mainnet.
+**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-docs/developer-tools/ide/playground/) on mainnet is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their canisters than on Ethereum.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:101]:
```suggestion
**ICP**: Currently, testing is done in the local environment. A [playground environment](/docs/current/developer-docs/developer-tools/ide/playground/) on the mainnet is available for canisters, but deployment time is limited to 20 minutes. However, the ability to upgrade and delete canisters on the mainnet allows developers greater flexibility to iterate and refine their smart contracts than on Ethereum.
```

File: docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx [hunk 31]
```diff
@@ -0,0 +1,124 @@
...
+
+#### Paying for gas
+
+**Ethereum**: Users pay for gas fees directly using the crypto wallet they connect with the dapp.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/ethereum/using-eth/eth-comparison.mdx:105]:
```suggestion
**Ethereum**: Users pay for gas fees directly using the crypto wallet they connect to the dapp.
```

**Pull request merged**